Unity/Assets/Scripts/BattleView.cs

```diff
@@ -0,0 +1,224 @@
+            { ZebraImage, zebra },
+            { DolphinImage, dolphin },
+            { OrcaImage, orca },
+            { HumanImage, human },
+        };
+        
+        Dictionary<Sprite, string> fields = new Dictionary<Sprite, string>()
+        {
+            { _fieldImages[(int)FieldConstant.Field.Tokyo], FieldConstant.Tokyo },
+            { _fieldImages[(int)FieldConstant.Field.Savannah], FieldConstant.Savannah },
+            { _fieldImages[(int)FieldConstant.Field.PacificOcean], FieldConstant.PacificOcean },
+            { _fieldImages[(int)FieldConstant.Field.Space], FieldConstant.Space },
+        };
+
+        Dictionary<string, Sprite> new_creatures_image = new Dictionary<string, Sprite>()
+        {
+            { ElephantImage, _characterBattleImages[(int)CreatureConstant.Creature.Elephant] },
+            { LionImage, _characterBattleImages[(int)CreatureConstant.Creature.Lion] },
+            { ZebraImage, _characterBattleImages[(int)CreatureConstant.Creature.Zebra] },
+            { DolphinImage, _characterBattleImages[(int)CreatureConstant.Creature.Dolphin] },
+            { OrcaImage, _characterBattleImages[(int)CreatureConstant.Creature.Orca] },
+            { HumanImage, _characterBattleImages[(int)CreatureConstant.Creature.Human] },
+        };
+
+        _playerCreatureBattleImage.sprite = new_creatures_image[_playerCreatureImage.sprite.name];
+        _opponentCreatureBattleImage.sprite = new_creatures_image[_opponentCreatureImage.sprite.name];
+
+        Creature player = creatures[_playerCreatureImage.sprite.name];
+        Creature opponent = creatures[_opponentCreatureImage.sprite.name];
+
+        string _field = fields[_fieldImage.sprite];
+
+        JudgeBattle(player, opponent, _field);
+    }
+
+    void JudgeBattle(Creature player, Creature opponent, string field)
+    {
+        Dictionary<string, int> fieldScale = new Dictionary<string, int>()
+        {
+            {FieldConstant.Tokyo, 15},
+            {FieldConstant.Savannah, 4},
+            {FieldConstant.PacificOcean, 6},
+            {FieldConstant.Space, 2}
+        };
+
+        if (player.field == field)
+        {
+            _playerScaledPower = player.power * fieldScale[player.field];
+        }
+        else if (field == FieldConstant.Space)
+        {
+            _playerScaledPower = player.power / fieldScale[FieldConstant.Space];
+        }
+        else 
+        {
+            _playerScaledPower = player.power;
+        }
+
+        if (opponent.field == field)
+        {
+            _opponentScaledPower = opponent.power * fieldScale[opponent.field];
+        }
+        else if (field == FieldConstant.Space)
+        {
+            _opponentScaledPower = opponent.power / fieldScale[FieldConstant.Space];
+        }
+        else 
+        {
+            _opponentScaledPower = opponent.power;
+        }
+
+        _playerCreatureText.text = player.field + "\n" + Power + ":" +
+            _playerScaledPower + "\n" + player.name;
```

Comment:
文字の連結は、+演算子だとメモリの確保が行われるため以下の方法をおすすめします。
```suggestion
        _playerCreatureText.text = $"{player.field}\n"{Power}:{_playerScaledPower}\n{player.name}";
```

参考: https://www.sejuku.net/blog/50133